nRFMeshProvision/Classes/Mesh Messages/Time and Scenes/TimeZoneStatus.swift

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (c) 2022, Nordic Semiconductor
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without modification,
+* are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+*    list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright notice, this
+*    list of conditions and the following disclaimer in the documentation and/or
+*    other materials provided with the distribution.
+*
+* 3. Neither the name of the copyright holder nor the names of its contributors may
+*    be used to endorse or promote products derived from this software without
+*    specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+* NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+* PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+* WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import Foundation
+
+public struct TimeZoneStatus: GenericMessage {
+    public static let opCode: UInt32 = 0x823D
+    
+    public var parameters: Data? {
+        let data =  Data() + currentTzOffset.encodeToTzOffset() + nextTzOffset.encodeToTzOffset()
+        
+        return data + Data([taiSeconds.getByte(at: 0), taiSeconds.getByte(at: 1), taiSeconds.getByte(at: 2), taiSeconds.getByte(at: 3), taiSeconds.getByte(at: 4)])
```

Comment:
This screams for improvement. Don't know how :)

Reply:
Then you're REALLY not going to like the PR I'm working on now, for scheduler. :)
I've been mulling over nicer ways to do this, and I'm thinking about doing something similar to Android Mesh SDK's "BitWriter" (and reader). Wrapping Data in a helper class to keep track of bit-offsets and sizes. There are a number of messages that do some rather serious bit-twiddling to minimize the package size.

Reply:
Using the "writeBits" extension on Data from the Scheduler PR. Still not really optimal, but I think it's a lot easier to understand what's going on at least.